spec2vec/vector_operations.py

```diff
@@ -60,3 +61,83 @@ def _check_model_coverage():
     weights_raised_tiled = numpy.tile(weights_raised, (1, model.wv.vector_size))
     vector = numpy.sum(word_vectors * weights_raised_tiled, 0)
     return vector
+
+
+@numba.njit
+def cosine_similarity_matrix(vectors_1: numpy.ndarray, vectors_2: numpy.ndarray) -> numpy.ndarray:
+    """Fast implementation of cosine similarity between two arrays of vectors.
+
+    For example:
+
+    .. code-block:: python
+
+        import numpy as np
+        from spec2vec.vector_operations import cosine_similarity_matrix
+
+        vectors_1 = np.array([[1, 1, 0, 0],
+                              [1, 0, 1, 1]])
+        vectors_2 = np.array([[0, 1, 1, 0],
+                              [0, 0, 1, 1]])
+        similarity_matrix = cosine_similarity_matrix(vectors_1, vectors_2)
+
+
+    Parameters
+    ----------
+    vectors_1
+        Numpy array of vectors. vectors_1.shape[0] is number of vectors, vectors_1.shape[1]
+        is vector dimension.
+    vectors_2
+        Numpy array of vectors. vectors_2.shape[0] is number of vectors, vectors_2.shape[1]
+        is vector dimension.
+    """
+    vectors_1 = vectors_1.astype(numpy.float64)  # Numba dot only accepts float or complex arrays
```

Comment:
Check whether inputs have same length
```suggestion
    assert vectors_1.shape[1] == vectors_2.shape[1], "Input vector must have same shape."
    vectors_1 = vectors_1.astype(numpy.float64)  # Numba dot only accepts float or complex arrays
```